File: DoubleDispatch/Event.cpp
/*
 * This code is released under GPLv2 License.
 *
 * http://www.gnu.org/licenses/gpl-2.0.txt
 *
 */

#include "Event.hpp"
#include "EventHandler.hpp"


void ApplicationEvent::accept(EventHandler *handler)
{ handler->handle(*this); }


void UserEvent::accept(EventHandler *handler)
{ handler->handle(*this); }

File: EventNotifier/eventnotifier/Event.java
/*
 * This code is released under GPLv2 License.
 *
 * http://www.gnu.org/licenses/gpl-2.0.txt
 *
 */

package eventnotifier;

public abstract class Event
{

    public static final int LOW = 1;
    public static final int MODERATE = 2;
    public static final int CRITICAL = 3;
    public int priority;

    Event()
    {
        priority = LOW;
    }

    Event(int aPriority)
    {
        if (aPriority < LOW) {
            priority = LOW;
        }
        else if (aPriority > CRITICAL) {
            priority = CRITICAL;
        }
        else {
            priority = aPriority;
        }
    }
}
